ns for fx: qat test cases for unshadowed activations

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56205 ns for fx: enable shadowing int8 to int8
* #56197 ns for fx: move unmatchable mod/fun/meth mapping to mappings file
* #56196 ns for fx: make `call_method` nodes work in NS APIs
* #56195 ns for fx: remove deprecated code
* #56194 ns for fx: enable `call_method` nodes in graph matching
* #55918 ns for fx: fill out coverage for node I/O types
* #55859 ns for fx: add category for ops which accept fp32 or int8 input
* #55858 ns for fx: move op dtype category mapping to separate file
* #55837 ns for fx: add test for op relationship coverage
* #55805 ns for fx: move pattern utils to separate file
* #55803 ns for fx: make get_reversed_fusions reuse quantization fusions
* #55614 ns for fx: add qat tests cases for shadowed activations
* **#55508 ns for fx: qat test cases for unshadowed activations**
* #55507 ns for fx: extend functional weight extraction testing to QAT
* #55506 ns for fx: add qat handling for weight extraction
* #55505 ns for fx: make NSTracer inherit from QuantizationTracer
* #55434 ns for fx: change node I/O determination to strict allowlist

Summary:

Adds QAT test cases for unshadowed activation APIs.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27650406](https://our.internmc.facebook.com/intern/diff/D27650406)

vkuzo committed:
ns for fx: qat test cases for unshadowed activations

Summary:

Adds QAT test cases for unshadowed activation APIs.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: qat test cases for unshadowed activations"


Summary:

Adds QAT test cases for unshadowed activation APIs.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27650406](https://our.internmc.facebook.com/intern/diff/D27650406)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_numeric_suite_fx.py:
```diff
@@ -474,10 +474,15 @@ def _test_match_activations(
         if qconfig_dict is None:
             qconfig_dict = {'': torch.quantization.default_qconfig}
-        mp = prepare_fx(m, qconfig_dict)
+        if prepare_fn == prepare_fx:
```
should we just check the mode of model to decide which prepare_fn to call?

jerryzh168 commented on test/quantization/test_numeric_suite_fx.py:
```diff
@@ -608,47 +613,51 @@ def test_extract_weights_dynamic(self):
 
-        m = M().eval()
+    def _test_match_activations_fun_impl(self, prepare_fn=prepare_fx):
+        m = LinearReluLinearFunctional().eval()
```
optional nit: I feel setting model to eval/train and use this to decide the rest might be more consistent with our apis

vkuzo committed:
rebase on "ns for fx: qat test cases for unshadowed activations"


Summary:

Adds QAT test cases for unshadowed activation APIs.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27650406](https://our.internmc.facebook.com/intern/diff/D27650406)

[ghstack-poisoned]

vkuzo committed:
rebase on "ns for fx: qat test cases for unshadowed activations"


Summary:

Adds QAT test cases for unshadowed activation APIs.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27650406](https://our.internmc.facebook.com/intern/diff/D27650406)

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: qat test cases for unshadowed activations"


Summary:

Adds QAT test cases for unshadowed activation APIs.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27650406](https://our.internmc.facebook.com/intern/diff/D27650406)

[ghstack-poisoned]
